### ISSUE TITLE
[🐛 Bug]:  Selenium::WebDriver::Error::NoSuchDriverError

### ISSUE THREAD

**Issue by JuzerShakir** [SeleniumHQ/selenium#13152]
### What happened?

I was previously running selenium `v4.7` using the Firefox browser with `capybara` library for testing with the following options:
```
Capybara.register_driver :headless_firefox do |app|
  options = Selenium::WebDriver::Remote::Capabilities.firefox(
    "moz:firefoxOptions": {args: %w[-headless -disable-gpu]}
  )
  Capybara::Selenium::Driver.new(app, browser: :firefox, options: options)
end

Capybara.javascript_driver = :headless_firefox
```
The following worked perfectly fine even in GitHub CI without downloading the Firefox browser or driver explicitly.

After updating my selenium gem to `v4.15`, I got the following error:
```
NoMethodError:
       undefined method `firefox' for Selenium::WebDriver::Remote::Capabilities:Class
```
I referred to [this documentation](https://www.selenium.dev/documentation/webdriver/browsers/firefox/) and made the changes accordingly:
```
Capybara.register_driver :headless_firefox do |app|
  options = Selenium::WebDriver::Options.firefox
  options.args << "--headless"

  Capybara::Selenium::Driver.new(app, browser: :firefox, options:)
end

Capybara.javascript_driver = :headless_firefox
```
And get the following error message:
```
Selenium::WebDriver::Error::NoSuchDriverError:
       Unable to obtain geckodriver using Selenium Manager; wrong number of arguments (given 2, expected 0..1);
```
To download the Firefox drivers, I navigated to [this repo](https://github.com/mozilla/geckodriver/releases) to download `v0.33` drivers for Linux `x64`. I added the executable file to `/usr/local/bin` path but it still returns the same error message.

I tried to change the browser option to `Chrome` but it returns the same error.

### How can we reproduce the issue?

```shell
git clone git@github.com:JuzerShakir/fmb.git

bundle install

rails db:create
rails db:migrate

rspec spec/features
```

### Relevant log output

```shell
Selenium::WebDriver::Error::NoSuchDriverError:
       Unable to obtain geckodriver using Selenium Manager; wrong number of arguments (given 2, expected 0..1); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/driver_finder.rb:30:in `rescue in path'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/driver_finder.rb:27:in `path'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/local_driver.rb:41:in `process_options'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/local_driver.rb:27:in `initialize_local_driver'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/firefox/driver.rb:42:in `initialize'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/driver.rb:53:in `new'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/driver.rb:53:in `for'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver.rb:89:in `for'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/capybara-3.38.0/lib/capybara/selenium/driver.rb:83:in `browser'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/capybara-3.38.0/lib/capybara/selenium/driver.rb:104:in `visit'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/capybara-3.38.0/lib/capybara/session.rb:280:in `visit'
     # /usr/share/rvm/gems/ruby-3.2.2/gems/rack_session_access-0.2.0/lib/rack_session_access/capybara.rb:6:in `set_rack_session'
     # ./spec/features/transactions/index_spec.rb:10:in `block (2 levels) in <main>'
     # ------------------
     # --- Caused by: ---
     # ArgumentError:
     #   wrong number of arguments (given 2, expected 0..1)
     #   /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/lib/selenium/webdriver/common/logger.rb:51:in `initialize'
```

### Operating System

Ubuntu 23.04 

### Selenium version

Ruby 4.15

### What are the browser(s) and version(s) where you see this issue?

Firefox

### What are the browser driver(s) and version(s) where you see this issue?

GeckoDriver v0.33

**Comment by titusfortner** [comment]
You need to also update your version of capybara. There was a bug in the way it wrapped the Selenium logger class.

**Comment by titusfortner** [comment]
And you shouldn't download the driver yourself, selenium will manage all of that now.

**Comment by JuzerShakir** [comment]
I updated my Capybara library from `v3.38.0` to `v3.39.2` and the error no longer shows up! Thanks a lot! :) 

But now I get this **error message** on running the tests with the Firefox browser:

![Screenshot from 2023-11-15 19-15-57](https://github.com/SeleniumHQ/selenium/assets/10782481/0f04533b-d628-48be-a7e1-ce10082d6e2e)

Setting up with Firefox browser:
```
Capybara.register_driver :firefox do |app|
  Capybara::Selenium::Driver.new(app, browser: :firefox)
end

Capybara.javascript_driver = :firefox
```

While setting up with Chrome browser, **it works perfectly fine**:
```
Capybara.register_driver :chrome do |app|
  Capybara::Selenium::Driver.new(app, browser: :chrome)
end

Capybara.javascript_driver = :chrome
```

**Comment by titusfortner** [comment]
Hmm, can you run your code with `Selenium::WebDriver.logger.level = :debug` and paste what you get?

**Comment by JuzerShakir** [comment]
Here's the log:
```shell
2023-11-16 10:16:24 INFO Selenium [:logger_info] Details on how to use and modify Selenium logger:
  https://selenium.dev/documentation/webdriver/troubleshooting/logging

2023-11-16 10:16:24 DEBUG Selenium [:selenium_manager] Selenium Manager binary found at /usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/bin/linux/selenium-manager 
2023-11-16 10:16:24 DEBUG Selenium [:selenium_manager] Executing Process ["/usr/share/rvm/gems/ruby-3.2.2/gems/selenium-webdriver-4.15.0/bin/linux/selenium-manager", "--browser", "firefox", "--output", "json", "--debug"] 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Found geckodriver 0.33.0 in PATH: /usr/local/bin/geckodriver 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] firefox detected at /usr/bin/firefox 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Running command: /usr/bin/firefox -v 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Output: "Mozilla Firefox 119.0.1" 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Detected browser: firefox 119.0.1 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Required driver: geckodriver 0.33.0 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Driver path: /usr/local/bin/geckodriver 
2023-11-16 10:16:27 DEBUG Selenium [:selenium_manager] Browser path: /usr/bin/firefox 
2023-11-16 10:16:27 DEBUG Selenium [:driver_service] Executing Process ["/usr/local/bin/geckodriver", "--port=4444"] 
2023-11-16 10:16:27 DEBUG Selenium [:process] Starting process: ["/usr/local/bin/geckodriver", "--port=4444"] with {[:out, :err]=>#<IO:<STDOUT>>, :pgroup=>true} 
2023-11-16 10:16:27 DEBUG Selenium [:process]   -> pid: 34889 
2023-11-16 10:16:27 DEBUG Selenium [:driver_service] polling for socket on ["127.0.0.1", 4444] 
1700109987064	geckodriver	INFO	Listening on 127.0.0.1:4444
2023-11-16 10:16:27 DEBUG Selenium [:command] -> POST session 
2023-11-16 10:16:27 DEBUG Selenium [:command]    >>> http://127.0.0.1:4444/session | {"capabilities":{"alwaysMatch":{"acceptInsecureCerts":true,"browserName":"firefox","moz:firefoxOptions":{"binary":"/usr/bin/firefox"},"moz:debuggerAddress":true}}} 
2023-11-16 10:16:27 DEBUG Selenium [:header]      > {"Accept"=>"application/json", "Content-Type"=>"application/json; charset=UTF-8", "User-Agent"=>"selenium/4.15.0 (ruby linux)", "Content-Length"=>"163"} 
1700109987553	mozrunner::runner	INFO	Running command: MOZ_CRASHREPORTER="1" MOZ_CRASHREPORTER_NO_REPORT="1" MOZ_CRASHREPORTER_SHUTDOWN="1" MOZ_NO_REMOTE="1" "/usr ... tte" "--remote-debugging-port" "9222" "--remote-allow-hosts" "localhost" "-no-remote" "-profile" "/tmp/rust_mozprofileTy86HP"
Gtk-Message: 10:16:27.718: Not loading module "atk-bridge": The functionality is provided by GTK natively. Please try to not load it.

(firefox:34987): Gtk-WARNING **: 10:16:27.808: GTK+ module /snap/firefox/3358/gnome-platform/usr/lib/gtk-2.0/modules/libcanberra-gtk-module.so cannot be loaded.
GTK+ 2.x symbols detected. Using GTK+ 2.x and GTK+ 3 in the same process is not supported.
Gtk-Message: 10:16:27.808: Failed to load module "canberra-gtk-module"

(firefox:34987): Gtk-WARNING **: 10:16:27.810: GTK+ module /snap/firefox/3358/gnome-platform/usr/lib/gtk-2.0/modules/libcanberra-gtk-module.so cannot be loaded.
GTK+ 2.x symbols detected. Using GTK+ 2.x and GTK+ 3 in the same process is not supported.
Gtk-Message: 10:16:27.810: Failed to load module "canberra-gtk-module"
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
```

**Comment by titusfortner** [comment]
So there is no error / stack trace in the code, just the warning that pops up in your system?

The Selenium bits look correct, so try to get more info from geckodriver:

I'm not certain about the Capybara parts, but should look like:
```
Capybara.register_driver :firefox do |app|
  service = Selenium::WebDriver::Service.firefox(log: $stdout, args: ['--log-level=TRACE'])
  Capybara::Selenium::Driver.new(app, browser: :firefox, service:)
end
```

**Comment by ka8725** [comment]
> Hmm, can you run your code with `Selenium::WebDriver.logger.level = :debug` and paste what you get?

I have the same issue with the description. Interestingly, it works with this setting `Selenium::WebDriver.logger.level = :debug`.

Full stack trace:

```ruby
     Selenium::WebDriver::Error::NoSuchDriverError:
       Unable to obtain chromedriver using Selenium Manager; wrong number of arguments (given 2, expected 0..1); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/driver_finder.rb:30:in `rescue in path'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/driver_finder.rb:27:in `path'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/local_driver.rb:41:in `process_options'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/local_driver.rb:27:in `initialize_local_driver'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/chrome/driver.rb:34:in `initialize'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/driver.rb:47:in `new'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/driver.rb:47:in `for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver.rb:89:in `for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/capybara-3.37.1/lib/capybara/selenium/driver.rb:83:in `browser'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/capybara-3.37.1/lib/capybara/selenium/driver.rb:104:in `visit'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/capybara-3.37.1/lib/capybara/session.rb:278:in `visit'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/capybara-3.37.1/lib/capybara/dsl.rb:52:in `call'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/capybara-3.37.1/lib/capybara/dsl.rb:52:in `visit'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-rails-5.1.2/lib/rspec/rails/example/feature_example_group.rb:29:in `visit'
     # ./spec/support/request_helpers.rb:10:in `login'
     # ./spec/features/users/projects/managing_a_project_with_messages_index_spec.rb:135:in `block (2 levels) in <top (required)>'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:365:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:529:in `block in run_owned_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:528:in `each'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:528:in `run_owned_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:615:in `block in run_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:614:in `reverse_each'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:614:in `run_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:484:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:505:in `run_before_example'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:261:in `block in run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:511:in `block in with_around_and_singleton_context_hooks'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:468:in `block in with_around_example_hooks'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:486:in `block in run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:626:in `block in run_around_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:352:in `call'
     # ./spec/features/users/projects/managing_a_project_with_messages_index_spec.rb:10:in `block (3 levels) in <top (required)>'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/climate_control-1.2.0/lib/climate_control.rb:24:in `block in modify'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/climate_control-1.2.0/lib/climate_control.rb:15:in `synchronize'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/climate_control-1.2.0/lib/climate_control.rb:15:in `modify'
     # ./spec/support/env_helpers.rb:7:in `with_modified_env'
     # ./spec/features/users/projects/managing_a_project_with_messages_index_spec.rb:10:in `block (2 levels) in <top (required)>'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:390:in `execute_with'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:628:in `block (2 levels) in run_around_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:352:in `call'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-rails-5.1.2/lib/rspec/rails/adapters.rb:75:in `block (2 levels) in <module:MinitestLifecycleAdapter>'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:457:in `instance_exec'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:390:in `execute_with'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:628:in `block (2 levels) in run_around_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:352:in `call'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:629:in `run_around_example_hooks_for'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/hooks.rb:486:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:468:in `with_around_example_hooks'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:511:in `with_around_and_singleton_context_hooks'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example.rb:259:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/ci/contrib/rspec/example.rb:45:in `block in run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/ci/test.rb:25:in `block in trace'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/tracing/tracer.rb:497:in `skip_trace'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/tracing/tracer.rb:136:in `trace'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/tracing.rb:18:in `trace'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/ci/test.rb:23:in `trace'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ddtrace-1.4.1/lib/datadog/ci/contrib/rspec/example.rb:31:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example_group.rb:646:in `block in run_examples'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example_group.rb:642:in `map'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example_group.rb:642:in `run_examples'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/example_group.rb:607:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:121:in `block (3 levels) in run_specs'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:121:in `map'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:121:in `block (2 levels) in run_specs'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/configuration.rb:2068:in `with_suite_hooks'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:116:in `block in run_specs'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/reporter.rb:74:in `report'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:115:in `run_specs'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:89:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:71:in `run'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/lib/rspec/core/runner.rb:45:in `invoke'
     # /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rspec-core-3.11.0/exe/rspec:4:in `<top (required)>'
     # ./bin/rspec:27:in `load'
     # ./bin/rspec:27:in `<main>'
     # ------------------
     # --- Caused by: ---
     # ArgumentError:
     #   wrong number of arguments (given 2, expected 0..1)
     #   /Users/ka8725/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/selenium-webdriver-4.16.0/lib/selenium/webdriver/common/logger.rb:51:in `initialize'
```


UPD. Setting the log level explicitly to info eventually made it work: `Selenium::WebDriver.logger.level = :info`

**Comment by titusfortner** [comment]
Oh, update Capybara. There was a fix in 3.39.1 for how it wraps the Selenium logger.
